include/zephyr/storage/flash_map.h

```diff
@@ -269,6 +269,48 @@ uint8_t flash_area_erased_val(const struct flash_area *fa);
 #define FLASH_AREA_SIZE(label) \
 	DT_REG_SIZE(DT_NODE_BY_FIXED_PARTITION_LABEL(label))
 
+/**
+ * Returns non-0 value if fixed-partition of given DTS node label exists.
+ *
+ * @param label DTS node label.
+ *
+ * @return non-0 if fixed-partition node exists and is enabled;
+ *	   0 if node does not exist, is not enabled or is not fixed-partition.
+ */
+#define FIXED_PARTITION_EXISTS(label) \
```

Comment:
Can you explain why you chose to use node labels instead of node identifiers? A node identifier would be more general and would allow you to refer to fixed partitions using any means you want: DT_CHILD(..., ...), DT_ALIAS(...), etc.

Reply:
> Can you explain why you chose to use node labels instead of node identifiers? A node identifier would be more general and would allow you to refer to fixed partitions using any means you want: DT_CHILD(..., ...), DT_ALIAS(...), etc.

It was easier.

I have change the macros to use nodes as proposed.